File: app/src/main/java/doit/study/droid/activities/SettingsActivity.java
package doit.study.droid.activities;

import android.os.Bundle;
import android.preference.PreferenceFragment;
;

import androidx.appcompat.app.AppCompatActivity;

import doit.study.droid.R;

public class SettingsActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getFragmentManager()
                .beginTransaction()
                .replace(android.R.id.content, new SettingsFragment())
                .commit();
    }

    public static class SettingsFragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            addPreferencesFromResource(R.xml.pref_general);
        }

    }
}

File: app/src/main/java/doit/study/droid/activities/SettingsActivity.kt
package doit.study.droid.activities

import android.os.Bundle
import android.preference.PreferenceFragment

import androidx.appcompat.app.AppCompatActivity

import doit.study.droid.R

class SettingsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fragmentManager
                .beginTransaction()
                .replace(android.R.id.content, SettingsFragment())
                .commit()
    }

    class SettingsFragment : PreferenceFragment() {
        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            addPreferencesFromResource(R.xml.pref_general)
        }

    }
}
